authoringAssistant/src/main/java/explorableviz/transparenttext/AuthoringAssistant.java

```diff
@@ -20,38 +21,53 @@ public AuthoringAssistant(LearningQueryContext learningQueryContext, String agen
         llm = initialiseAgent(agentClassName);
     }
 
-    public String execute(QueryContext query) throws Exception {
+    public AuthoringAssistantResult execute(QueryContext query) throws Exception {
         String response = null;
         int limit = Settings.getInstance().getLimit();
         // Add the input query to the KB that will be sent to the LLM
         PromptList sessionPrompt = (PromptList) prompts.clone();
-        sessionPrompt.addPrompt(PromptList.USER, query.toUserPrompt());
-        for (int attempts = 0; response == null && attempts <= limit; attempts++) {
+
+        int attempts;
+        long start = System.currentTimeMillis();
+        if (query.toUserPrompt().contains("REPLACE value=\\\"?") && Settings.getInstance().isReasoningEnabled()) {
```

Comment:
The `Settings.getInstance().getValue()` pattern is a bit verbose. Since the user has to go via a static method anyway, why not make each property static (and just internally use the single instance)? I.e. just provide `Settings.getValue()` for each value.